### PR TITLE
Adding TypeDoc comments for Auth methods.

### PR DESCRIPTION
**Pull request by egilmorez**
Adding TypeDoc comments for Auth methods, along with related TOC tweaks.

Staged internally at: https://firebase.devsite.corp.google.com/docs/reference/admin/node/admin.auth.Auth

Thanks!

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -393,241 +415,1348 @@ declare namespace admin.app {
...
+     * Returns the list of existing provider configurations matching the filter
+     * provided.
+     * At most, 100 provider configs can be listed at a time.
+     *
+     * SAML and OIDC provider support requires Google Cloud's Identity Platform
+     * (GCIP). To learn more about GCIP, including pricing and features,
+     * see the [GCIP documentation](https://cloud.google.com/identity-cp).
+     *
+     * @param options The provider config
+     *     filter to apply.
```
Comment (bojeil-google) [inline src/index.d.ts]: move to previous line

File: src/index.d.ts [hunk 2]
```diff
@@ -393,241 +415,1348 @@ declare namespace admin.app {
...
+     * @return A promise that
```
Comment (bojeil-google) [inline src/index.d.ts]: Next line can probably fit on this line

File: src/index.d.ts [hunk 3]
```diff
@@ -393,241 +415,1348 @@ declare namespace admin.app {
...
+     * The next page token, if available.
+     */
     pageToken?: string;
   }
 
+  
   type UpdateAuthProviderRequest =
       admin.auth.SAMLUpdateAuthProviderRequest | admin.auth.OIDCUpdateAuthProviderRequest;
-
+
```
[the remark this answers is not in the document]
Comment (egilmorez) [reply src/index.d.ts]: Two space back resolves it?  Or should there be no indent at all -- just one line?  Thanks!

File: src/index.d.ts [hunk 4]
```diff
@@ -393,241 +415,1348 @@ declare namespace admin.app {
...
+      *     the app if it is installed.
+      *     If the actionCodeSettings is not specified, no URL is appended to the
+      *     action URL.
+      *     The state URL provided must belong to a domain that is whitelisted by the
+      *     developer in the console. Otherwise an error is thrown.
+      *     Mobile app redirects are only applicable if the developer configures
+      *     and accepts the Firebase Dynamic Links terms of service.
+      *     The Android package name and iOS bundle ID are respected only if they
+      *     are configured in the same Firebase Auth project.
+      * @return
```
[the remark this answers is not in the document]
Comment (egilmorez) [reply src/index.d.ts]: I think there are a few " @return {!Promise<string>}" in the externs that didn't have any further explanation.  I will not exercise my right to claim they are out of scope :)

File: src/index.d.ts [hunk 5]
```diff
@@ -393,241 +415,1338 @@ declare namespace admin.app {
...
+       * The Android minimum version if available. If the installed app is an older
+       * version, the user is taken to the GOogle Play Store to upgrade the app.
+       */
       minimumVersion?: string;
     };
+  
+    /**
+     * Defines the dynamic link domain to use for the current link if it is to be
+     * opened using Firebase Dynamic Links, as multiple dynamic link domains can be
+     * configured per project. This field provides the ability explicitly choose
```
Comment (bojeil-google) [inline src/index.d.ts]: This fields provides the ability to explicitly choose...

**Pull request merged**